TaskLayer/XLSearchTask/XLSearchTask.cs

```diff
@@ -145,7 +145,23 @@ protected override MyTaskResults RunSpecific(string OutputFolder, List<DbForTask
                 filteredAllPsms.Add(RemoveDuplicateFromPsmsPerScan(psmsPerScan));
             }
 
-            var allPsmsList = filteredAllPsms.Select(p => p.First()).OrderByDescending(p => p.XLTotalScore).ToList();
+            var allPsmsList = new List<CrosslinkSpectralMatch>();
+
+            foreach (var psmsPerScan in allPsms)
```

Comment:
Add comments to explain, I don't understand what is happening

Reply:
Comment added